src/bootstrap/config.rs

```diff
@@ -1380,21 +1380,34 @@ impl Config {
         git
     }
 
-    pub(crate) fn artifact_channel(&self, builder: &Builder<'_>, commit: &str) -> String {
-        if builder.rust_info.is_managed_git_subrepository() {
+    pub(crate) fn artifact_version_part(&self, builder: &Builder<'_>, commit: &str) -> String {
```

Comment:
```suggestion
    /// Bootstrap embeds a version number into the name of shared libraries it uploads in CI. 
    /// Return the version it would have used for the given commit.
    pub(crate) fn artifact_version_part(&self, builder: &Builder<'_>, commit: &str) -> String {
```